docs/tutorials/package-role-migration.md

```diff
@@ -0,0 +1,139 @@
+
+```sh
+yarn backstage-cli migrate package-scripts
+```
+
+The migration command also carries over any existing flags that were being passed in the old scripts.
+
+If you in the end do not want to use this exact script setup, it is still recommended to migrate to using the `package` commands, as the top-level commands will be deprecated and removed. If you don't want to use package roles either, you can pass an explicit role to some of the package commands, for example `yarn backstage-cli package build --role web-library`.
+
+### Step 3 - Migrate package ESLint configurations
+
+An area that has been simplified as part of the move to package roles is the ESLint configuration. Rather than having each package select which configuration they want (and getting it wrong), they now use a shared configuration factory that utilizes the package role. You can read more about the new configuration setup in the [build system documentation](../local-dev/cli-build-system#linting).
+
+To migrate the ESLint configuration of all packages in your project, run the following command:
+
+```sh
+yarn backstage-cli migrate package-lint-configs
+```
+
+This will migrate all existing `.eslintrc.js` that extend the old configuration from `@backstage/cli`, as well as carry over any additional configuration.
+
+### Step 4 - Use `backstage-cli repo`
+
+The Backstage CLI recently introduced a new `repo` command category, which houses commands that operate on an entire monorepo at once. These commands work particularly well once packages have been migrated to use roles, as that allows for some very effective optimizations. It is typically much faster to use these commands compared to using tools like `lerna`, as they're able to avoid the overhead of calling package scripts through `yarn` and can operate on multiple packages at once. You can read more about the `repo` command in the [CLI command documentation](../local-dev/cli-commands#repo-build).
+
+The way to execute this step of the migration is not as well defined as the previous steps, as it depends on what your development and CI/CD setup looks like. Look for the following patterns to replace in your root `package.json` as well as CI/CD setup:
+
+- Commands that lint the entire repo should be replaced with `yarn backstage-cli repo lint` along with a `--since` flag if needed. For example this:
+
+  ```sh
+  lerna run lint --since origin/master --
+  ```
+
+  would be replaced by the following:
+
+  ```sh
+  backstage-cli repo lint --since origin/master
+  ```
+
+- In places where the entire repo is being built, use `yarn backstage-cli repo build`, which also supports the `--since` flag. The migration here is a bit more nuanced as it depends why you are building all packages.
```

Comment:
```suggestion
- In places where the entire repo is being built, use `yarn backstage-cli repo build`, which also supports the `--since` flag. The migration here is a bit more nuanced as it depends on why you are building all packages.
```

Signed-off-by: Fredrik Adelöw <freben@gmail.com>